configuration/vpn.md

```diff
@@ -60,73 +60,128 @@ Using a ProxyVM to set up a VPN client gives you the ability to:
-
-    In the following, we use the following placeholder:  
-    `$DEV`  For the device created for the connection.  
-    `$PROT` For the protocol used for connection  
-    `$SVR`  For the openvpn server's ip.  
-    `$DNS`  For the dns server's ip.  
-
+    Copy your openvpn config files to `/rw/config/openvpn/` folder. The example main config file is `openvpn-client.ovpn`.
 
-3.  Setup iptables:  
-    Edit `/rw/config/qubes-firewall-user-script` and add:
+    It should have one line that reads `dev tun`.
 
-    `iptables -P OUTPUT DROP`  
-    This blocks all outgoing traffic, if not specified otherwise.
+    If it does not contain a line `redirect-gateway def1` you may wish to add it. This will route all traffic through your vpn's network device after a connection is created. However, many VPN services will push this instruction to your client automatically -- having a line that says `client` or `pull` in your openvpn config instructs your client to use parameters specified by the VPN server.
     
-    `iptables -I OUTPUT -o $DEV -j ACCEPT`  
-    This allows the local system to connect through the vpn (you dont need this).
-    
-    `iptables -I OUTPUT -o eth0 -d $SVR -p $PROT --dport 1194 -j ACCEPT`  
-    This allows your system to connect to the vpn server with the protocol `$PROT` under the port 1194.
-    
-    `iptables -I OUTPUT -o lo -j ACCEPT`  
-    This allows connections from the system to the system.
-
-    `iptables -I FORWARD -o eth0 -j DROP`  
-    `iptables -I FORWARD -i eth0 -j DROP`  
-    This blocks forwarding of connections through your plain network device (in case the vpn tunnel breaks).
-
-    `iptables -t nat -I PR-QBS -p udp --dport 53 -j DNAT --to-destination $DNS`  
-    `iptables -t nat -I PR-QBS -p tcp --dport 53 -j DNAT --to-destination $DNS`  
-    This will rewrite the DNS destination, and the traffic will be routed down the vpn tunnel. (to prevent DNS leaks)
-
+    NOTE: If the connection breaks down all traffic will by default be routed through the upstream network device eth0 (we will stop this with iptables in step 3).
+
+    Also add the following to accomodate a DNS script:
+    ```
+    script-security 2
+    up 'qubes-vpn-handler.sh up'
+    down 'qubes-vpn-handler.sh down'
+    ```
+
+3.  Setup iptables.
+    Edit the firewall script with `sudo nano /rw/config/qubes-firewall-user-script` and add:
+
+	```
+	#!/bin/bash
+	#    First, block all outgoing traffic
+	iptables -P OUTPUT DROP
+    iptables -F OUTPUT
+
+	#    Add the `qvpn` group to system, if it doesn't already exist
+    if ! grep -q "^qvpn:" /etc/group ; then
+        groupadd -rf qvpn
+        sync
+    fi
+    sleep 2s
+
+	#    Allow traffic from the `qvpn` group to the uplink interface (eth0);
+	#    Our openvpn will run as group `qvpn`.
+    iptables -A OUTPUT -p all -o eth0 -m owner --gid-owner qvpn \
+    -m state --state NEW,ESTABLISHED -j ACCEPT
```

Comment:
Any particular reason for including state when both NEW and ESTABLISHED are included? If you want to exclude INVALID or sth like this, better write to it explicitly with `! --state INVALID`. Otherwise the rule can be simplified to:
`iptables -A OUTPUT -o eth0 -m owner --gid-owner qvpn -j ACCEPT`